storage/cloud.py

```diff
@@ -1018,15 +1024,6 @@ def get_direct_upload_url(self, path, mime_type, requires_cors=True):
 
         return super(RadosGWStorage, self).get_direct_upload_url(path, mime_type, requires_cors)
 
-    def complete_chunked_upload(self, uuid, final_path, storage_metadata):
```

Comment:
To my knowledge the RADOS driver is used as a catch all for other storage engines that aren't explicitly supported, won't this break for users using this driver for storage engines that don't support multipart uploads? FWIW this probably isn't a very common scenario.

Reply:
Added a compatiblity switch to the driver: if `multipart_upload: true` then we force server side assembly and push of the end file in chunks of 32 MB (or whatever the user tries). If `multipart_upload: false`, we fall back to client side assembly, but in order to accomodate big layer pushes, we increase `boto` timeout to 600 seconds (instead of default 60).